code/controllers/subsystem/ships.dm

```diff
@@ -78,9 +78,9 @@ var/global/list/ftl_weapons_consoles = list()
 			return faction_A.relations[i]
 
 
-/datum/subsystem/ship/proc/calculate_damage_effects(var/datum/starship/S)
-
-	S.fire_rate = round(initial(S.fire_rate) * factor_damage_inverse(SHIP_WEAPONS,S))
+/datum/subsystem/ship/proc/calculate_damage_effects(var/datum/starship/S, var/datum/component/weapon/W)
+	for(W in S)
+		W.fire_rate = round(initial(W.fire_rate) * factor_damage_inverse(SHIP_WEAPONS,S))
```

Comment:
You take an input weapon then use it in a for loop for ships, wtf

should be for(var/datum/component/weapon/W in S.components)

and take out the input var/datum/component/weapon/W